presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayFilterFunction.java

```diff
@@ -97,8 +97,13 @@ public static Block filter(Type type, Block block, MethodHandle function)
                 throw Throwables.propagate(throwable);
             }
             if (TRUE.equals(keep)) {
-                block.writePositionTo(position, resultBuilder);
-                resultBuilder.closeEntry();
+                if (block.isNull(position)) {
```

Comment:
You got it. That's why I thought `writePositionTo` should suffice when I wrote it. I remember there's a way to just copy and not worried about nulls. I thought `writePositionTo` was the one.